subsys/bootloader/include/fprotect.h

```diff
@@ -10,7 +10,18 @@
 #include <stdint.h>
 #include <string.h>
 
-/* Flash protect specified area */
-void fprotect_area(u32_t start, size_t length);
+/**
+ * @brief Protect flash area against writes.
+ *
+ * @param[in]  start   Start of range to protect.
+ * @param[in]  length  Length in bytes of range to protect.
+ *
+ * @retval 0        On success.
+ * @retval -EFAULT  If any of the argument are incorrect.
+ * @retval -EAGAIN  If function is called too many times. Applies to
```

Comment:
`EAGAIN` signals that the operation can be retried. If this is not the case here, consider using another error.